mitodl/open-discussions
# stop building individual ts packages

#### Pre-Flight checklist


#### What are the relevant tickets?
#3691 

#### What's this PR do?
This PR changes how we build Open Discussions.
- **Previously:** We had several local typescript packages (`ol-widgets`, etc) that were used by the apps (`infinite-corridor` aka "new open" and `open-discussions` aka "old open"). These individual TS packages were built from TS to JS for consumption in old open. This had some downsides:
    - We had been running 1 TS watch process per package. Doesn't seem great.
    - It complicated the webpack build... We needed to ensure that all TS packages were build prior to building webpack. This is fairly simple when actually building (just pass `--topological` to yarn's foreach command). But it was trickier for dev mode, where the webpack dev servers would try to build, but the inidviudal TS builds might not have completed yet. (Or one `ol-widgets` might try to build itself before its dependency, `ol-util`, was done building). We had previously "solved" this building **before** entering dev mode. But that's extra work.  
  
- **Now:** We do not build Typescript directly: `ol-widgets`, etc, stay as typescript files. The consuming app (just `infinite-corridor`) compiles the typescript to JS as part of the webpack build. _This has become simpler after #3862 _

#### How should this be manually tested?
1. Run `docker compose up` and separately watch the `watch` logs: `docker compose logs --follow --no-log-prefix  --tail 1000 watch`. You should see `open-discussions` and `infinite-corridor` enter dev mode, and they should be done building fairly quickly (For me, about 10s for infinite and 30s for open).
2. Make a change to one of the TS packages (e.g., add a `console.log("Hi")` to some file. Webpack should rebuild itself. should build.
